framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -119,279 +82,156 @@ private static Pattern anchored(String regex) {
      * @param localScope path to local scope to use
      * @param useLocalScope whether {@code localScope} should be used to resolve identifiers
      */
-    public static FlowExpressions.Receiver parse(
+    public static Receiver parse(
             String expression,
             FlowExpressionContext context,
             TreePath localScope,
             boolean useLocalScope)
             throws FlowExpressionParseException {
         context = context.copyAndSetUseLocalScope(useLocalScope);
-        FlowExpressions.Receiver result = parseHelper(expression, context, localScope);
+        Expression expr;
+        try {
+            expr = StaticJavaParser.parseExpression(noHashTags(expression));
+        } catch (ParseProblemException e) {
+            throw constructParserException(expression, "is an invalid expression");
+        }
+        Receiver result = parseHelper(expr, context, localScope);
         if (result instanceof ClassName && !expression.endsWith("class")) {
             throw constructParserException(
                     expression, "a class name cannot terminate a flow expression string");
         }
         return result;
     }
 
-    private static FlowExpressions.Receiver parseHelper(
-            String expression, FlowExpressionContext context, TreePath path)
+    private static Receiver parseHelper(
```

Comment:
This method is no longer "parsing" it's converting JavaParser Expressions to FlowExpression.Receivers. The method name (and those that it calls) should reflect that.